fv3_cap.F90

```diff
@@ -0,0 +1,1691 @@
+    call ESMF_TimeIntervalSet(timeStep,s=dt_atmos,rc=rc)
+    call ESMF_ClockSet(clock_fv3,timeStep=timeStep, rc=rc)
+!
+!------------------------------------------------------------------------
+! may need to set currTime for restart
+!
+    call ESMF_ClockGet(clock_fv3, currTIME=CurrTime,  StartTime=startTime,    &
+                       RunDuration=RunDuration, rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+
+    StopTime = startTime + RunDuration
+
+! *** read restart time from restart file
+    do i=751,899
+       inquire(i, opened=OPENED)
+       if(.not. OPENED)then
+         io_unit = i
+         exit
+       endif
+    enddo
+!
+    date = 0 ; date_init = 0
+    force_date_from_configure = .true.
+!
+    open(unit=io_unit, file=trim('INPUT/coupler.res'),status="old",err=998 )
+    read (io_unit,*,err=999) calendar_type
+    read (io_unit,*) date_init
+    read (io_unit,*) date
+    close(io_unit)
+    force_date_from_configure = .false.
+!
+    if(date(1) == 0 .and. date_init(1) /= 0) date = date_init
+    if(mype == 0) print *,'bf clock_fv3,date=',date,'date_init=',date_init
+
+    call ESMF_VMbroadcast(vm, date, 6, 0)
+    call ESMF_TimeSet(time=CurrTime,yy=date(1),mm=date(2),dd=date(3),h=date(4), &
+                      m=date(5),s=date(6),rc=rc)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+999 continue
+998 continue
+!    if(mype==0) print *,'final date =',date,'date_init=',date_init
+
+!reset CurrTime in clock
+    call ESMF_ClockSet(clock_fv3, currTIME=CurrTime, startTime=startTime,  &
+                       stopTime=stopTime, timeStep=timeStep, rc=rc)
+
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, line=__LINE__, file=__FILE__)) return
+ 
+    !
+    !Under NUOPC, the EARTH driver clock is a separate instance from the
+    ! - fv3 clock. However, the fv3 clock may have been reset from restart
+    ! - therefore the EARTH driver clock must also be adjusted.
+    ! - Affected: currTime, timeStep
+    call ESMF_ClockGet(clock, timeStep=earthStep, rc=RC)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+      line=__LINE__, file=__FILE__)) call ESMF_Finalize(endflag=ESMF_END_ABORT)
+
+    if (earthStep > (stopTime-currTime)) earthStep = stopTime - currTime
+    call ESMF_ClockSet(clock, currTime=currTime, timeStep=earthStep, rc=RC)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+      line=__LINE__, file=__FILE__)) call ESMF_Finalize(endflag=ESMF_END_ABORT)
+
+    ! Set fv3 component clock as copy of EARTH clock.
+    call NUOPC_CompSetClock(gcomp, clock, rc=RC)
+    if (ESMF_LogFoundError(rcToCheck=rc, msg=ESMF_LOGERR_PASSTHRU, &
+      line=__LINE__, file=__FILE__)) call ESMF_Finalize(endflag=ESMF_END_ABORT)
+
+    ! Read in the FV3 coupling interval
+    if ( cpl ) then
+      call clock_cplIntval(gcomp, CF)
```

Comment:
see module_cap_cpl

Reply:
This reads `atm_coupling_interval_sec:` from the config and sets the clock timeStep.

Reply:
what does this function actually do?

it reads `atm_coupling_interval_sec`, gets the clock from gcomp, and sets the timeStep of that clock (local variable) based on atm_coupling_interval_sec. Will gcomp's clock be changed?